git_machete/tests/functional/test_machete.py

```diff
@@ -36,110 +37,128 @@ def __init__(self) -> None:
         self.opt_yes: bool = True
 
 
-class GitAPIState:
-    pulls: List[Dict[str, Any]] = [{'head': {'ref': 'bugfix/remove'}, 'user': {'login': 'github_user'}, 'base': {'ref': 'develop'}, 'number': '15', 'html_url': 'www.github.com'}]
-    user: Dict[str, str] = {'login': 'github_user', 'type': 'User', 'company': 'VirtusLab'}
-    issues: List[Dict[str, Any]] = []
-    remote_branches: List[str] = []
+class MockGithubAPIState:
+    def __init__(self, pulls: List[Dict[str, Any]]) -> None:
+        self.pulls: List[Dict[str, Any]] = pulls
+        self.user: Dict[str, str] = {'login': 'github_user', 'type': 'User', 'company': 'VirtusLab'}
+        self.issues: List[Dict[str, Any]] = []
+
+    def new_request(self) -> "MockGithubAPIRequest":
+        return MockGithubAPIRequest(self)
+
+    def get_issue(self, issue_no: str) -> Optional[Dict[str, Any]]:
+        for index, issue in enumerate(self.issues):
+            if issue['number'] == issue_no:
+                return self.issues.pop(index)
+        return None
+
+    def get_pull(self, pull_no: str) -> Optional[Dict[str, Any]]:
+        for index, pull in enumerate(self.pulls):
+            if pull['number'] == pull_no:
+                return self.pulls.pop(index)
```

Comment:
Same here, why remove the pull request?

Reply:
It was used for update only where i had to replace given pull/issue anyway. Right now for possible future usage, ` get_pull/issue` doesn't remove item from collection.